### DISCUSSION TITLE
Reduce the size of the library

### DISCUSSION THREAD

**Discussion by MiOiM** [tttapa/Control-Surface#644] [Q&A]
Hi guys.

I have run out of flash storage, and still have to implement more code to my sketch. 
Is there a way to somehow reduce the size of the Control Surface library?
Sorry if it's a stupid question, I'm not a developer.

I'm using a PRO MICRO, with a SSD1306 screen, a few buttons and a few leds only, but I'm trying to customize it to use it as a pedalboard to control my daw/amp simulator vst.

Thank you!

**Comment by Veroledez** [comment]
Hola, yo abandone el leonardo por problemas de memoria, si usas oled te consume mucha memoria, tttapa tiene un repositorio optimizado de la libreria para el oled,   https://github.com/tttapa/Adafruit_SSD1306

La optimizacion es algo dificil, si te quedas sin espacio tendrias que usar un arduino  con mas memoria, como un teensy  que trae el usb nativo, si donde vives es muy costoso como en mi caso que un teensy vale por 6 leonardos  o pro micros, usa un esp8266 comunicado por serial midi  al pro micro y que el esp8266 haga todo, potencia y espacio le sobra

**Comment by tttapa** [comment]
There is no simple way to save memory without disabling or editing significant parts of the library. It's probably a better idea to get a board with more memory. There are boards that are cheaper than the pro micro but with many times the memory (both RAM and flash). Especially when using a display, you'll benefit from a board with more memory. Personally, I'd recommend a Teensy.

> tttapa tiene un repositorio optimizado de la libreria para el oled, https://github.com/tttapa/Adafruit_SSD1306

This will mostly save RAM, and it won't help much if you're just using a single display.
